### PR TITLE
Allow committing empty segment when some messages are fetched but all filtered out

### LINKED ISSUE #9014: Allowing empty segments for all stream consumers
In this part of the code, https://github.com/apache/pinot/blob/2b82366c369cf668175695605f0b77d7105f4aff/pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java#L301-L304, we extend the segment build time by 1 hour if no rows were consumed. We have some common cases where certain partitions may never have data. We are either filtering out a lot of the data or there's a temporary error in our ingestion plugin. We also use a custom ingestion plugin.

But by forever extending the lease like this, whenever we restart the server, the server attempts to consume going back days or weeks and causes high CPU usage and throttling for both the server and kafka.

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Fix #9014

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -573,12 +574,13 @@ private void processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeS
       streamMessageCount++;
     }
     updateCurrentDocumentCountMetrics();
-    if (streamMessageCount != 0) {
```
Comment (xiangfu0) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:576]: Does the behavior consistent for other streaming solutions like kinesis?

Comment (xiangfu0) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:576]: Also will this behavior change metrics reporting

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:576]: Yes, this part is shared across all streams.
The unfiltered message count is always >= message count, so the existing behavior is preserved

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 2]
```diff
@@ -243,6 +243,7 @@ public void deleteSegmentFile() {
 
   // Segment end criteria
   private volatile long _consumeEndTime = 0;
+  private volatile boolean _hasMessagesFetched = false;
```
Comment (sajjad-moradi) [inline pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:246]: nit: `hasFetchedAnyMessages` or  just`hasFetchedMessages`?

**Review by sajjad-moradi — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java:246]: I prefer the current name to be consistent with other variables, e.g. `_numRowsConsumed`, `_forceCommitMessageReceived` etc.

**Pull request merged**